plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1897,24 +1894,24 @@ public void testDynamicFilteringWithAggregationGroupingColumn()
                 PARTITIONED);
     }
 
-    @Flaky(issue = "https://github.com/trinodb/trino/issues/18499", match = ".*SqlQueryManager.getFullQueryInfo.*")
     @Test
     public void testDynamicFilteringWithAggregationAggregateColumn()
     {
         skipTestUnless(hasBehavior(SUPPORTS_DYNAMIC_FILTER_PUSHDOWN));
-        MaterializedResultWithQueryId resultWithQueryId = getDistributedQueryRunner()
-                .executeWithQueryId(getSession(), "SELECT custkey, count(*) count FROM orders GROUP BY custkey");
-        // Detecting whether above aggregation is fully pushed down explicitly as there are cases where SUPPORTS_AGGREGATION_PUSHDOWN
-        // is false as not all aggregations are pushed down but the above aggregation is.
-        boolean isAggregationPushedDown = getPhysicalInputPositions(resultWithQueryId.getQueryId()) == 1000;
-        assertDynamicFiltering(
-                "SELECT * FROM (SELECT custkey, count(*) count FROM orders GROUP BY custkey) a JOIN orders b " +
-                        "ON a.count = b.custkey AND b.totalprice < 1000",
-                PARTITIONED,
-                isAggregationPushedDown);
+        executeExclusively(() -> {
+            MaterializedResultWithQueryId resultWithQueryId = getDistributedQueryRunner()
```

Comment:
please extract methods instead of long lambdas. The diff will be more human friendly.

Reply:
For extracted methods I'll add `Flaky` suffix for existing method names

Reply:
Please don't. `Flaky` "means" non deterministic. Whereas we know how it behaves and it is deterministic. I think the better name would be `Unsafe` which is common term in regards to concurrency. However here I think a code comment would be simply the best.

Reply:
Method signatures are the same as of original methods, so I've used `Unsafe` suffix.